queue-manager/clear-queue-manager.go

```diff
@@ -58,13 +60,40 @@ func (handler *ClearOperationHandlerT) Exec(payload []byte) error {
 		},
 	}
 
+	var pm processor.ProcessorManagerI
+	for {
+		pm, err = processor.GetProcessorManager()
+		if err == nil {
+			break
+		}
+		pkgLogger.Infof("ProcessorManager is nil. Retrying after a second")
+		time.Sleep(time.Second)
+	}
+	pm.Pause()
+
+	var rm router.RoutersManagerI
+	for {
+		rm, err = router.GetRoutersManager()
+		if err == nil {
+			break
+		}
+		pkgLogger.Infof("RoutersManager is nil. Retrying after a second")
+		time.Sleep(time.Second)
+	}
+	rm.PauseAll()
+
+	//TODO remove
+	time.Sleep(15 * time.Second)
+
 	//Clear From GatewayDB
 	handler.clearFromJobsdb(clearOperationHandler.gatewayDB, parameterFilters, false, false)
```

Comment:
Instead of doing them in order GetRetry etc, which results in three queries, how about get non-terminal states and mark them aborted. Similar to what we do for cluster migration.

Reply:
done